WalletWasabi.Tests/StoreTests/MempoolTests.cs

```diff
@@ -0,0 +1,103 @@
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using WalletWasabi.Helpers;
+using WalletWasabi.Models;
+using WalletWasabi.Stores.Mempool;
+using Xunit;
+
+namespace WalletWasabi.Tests.StoreTests
+{
+	public class MempoolTests
+	{
+		[Fact]
+		public async Task CanInitializeMempoolStoreAsync()
+		{
+			var mempoolStore = new MempoolStore();
+
+			var dir = Path.Combine(Global.Instance.DataDir, EnvironmentHelpers.GetMethodName());
+			var network = Network.Main;
+			await mempoolStore.InitializeAsync(dir, network, false);
+
+			Assert.Equal(network, mempoolStore.Network);
+			Assert.Equal(dir, mempoolStore.WorkFolderPath);
+		}
+
+		[Fact]
+		public async Task CanDoOperationsAsync()
+		{
+			var mempoolStore = new MempoolStore();
+
+			var dir = Path.Combine(Global.Instance.DataDir, EnvironmentHelpers.GetMethodName());
+			await IoHelpers.DeleteRecursivelyWithMagicDustAsync(dir);
+			var network = Network.Main;
+			await mempoolStore.InitializeAsync(dir, network, false);
+
+			Assert.True(mempoolStore.IsEmpty());
+
+			var tx = Transaction.Create(network);
+			var stx = new SmartTransaction(tx, Height.Mempool, firstSeenIfMempoolTime: DateTimeOffset.UtcNow);
+			Assert.True(mempoolStore.TryAdd(stx.GetHash()));
+			var isAdded = mempoolStore.TryAdd(stx);
+			Assert.False(isAdded.isHashAdded);
+			Assert.True(isAdded.isTxAdded);
```

Comment:
No, not in this function. But if you'd only return a boolean you wouldn't know if that'd mean both the txhash and the tx is added or just the tx is added and the txhash wasn't (because it was already present.)